continuous_integration/flake8_diff.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+
+set -e
+
+# Travis does the git clone with a limited depth (50 at the time of
+# writing). This may not be enough to find the common ancestor with
+# $REMOTE/master so we unshallow the git checkout
+git fetch --unshallow || echo "Unshallowing the git checkout failed"
```

Comment:
Do you have an idea of this performance impact of a full fetch on a project like joblib and on a large project like scikit-learn?